test/typescript/dispatch.ts

```diff
@@ -9,4 +9,4 @@ const dispatchResult: Action = dispatch({type: 'TYPE'});
 
 type Thunk<O> = () => O;
 
-const dispatchThunkResult: number = dispatch(() => 42);
+const dispatchThunkResult: number = dispatch<Thunk<number>, number>(() => 42);
```

Comment:
@Pajn , the point of generics is to provide typings over something that is user-defined, in this case the action and the result. While being able to infer them would be ideal, it is still just a convenience.

> Could be written

The argument is purely cosmetic and the increase of readability is arguable. It may sound a bit harsh, but which example being shorter is irrelevant, because that's not our goal. Otherwise we would be writing pure JavaScript instead. The main point here is that we end up with something sane, instead of an implicit `any` that can just leak all over the application. I shot myself in the foot more than a few times with loose definitions and I can tell you that it's better to have:

```typescript
getSomething<T>(): T;
```
than
```typescript
getSomething(): any;
```

While neither guarantees that you actually cast to the correct type, the first one at least ensures that you made a conscious decision, instead of accepting `any` by default.